File: Models/HubMessages/Fuji/FujiUpdate.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Models.HubMessages.Fuji
{
    public class FujiUpdate
    {
        public List<PlayedCardUpdate> PlayedCards { get; set; }
        public List<PlayerDiscardUpdate> DiscardedCards { get; set; }
        public List<PlayerDrawUpdate> NewDraws { get; set; }
        public ActivePlayerUpdate ActivePlayerUpdate { get; set; }
        public GameVictoryUpdate GameVictoryUpdate { get; set; }
    }
}

File: GooseGames/ClientApp/src/models/fujiflush/fujiupdate.ts

export class FujiPlayedCardUpdate {
  playerId: string;
  faceValue: number;
  combinedValue: number;
}


export class FujiDiscardUpdate {
  playerId: string;
}



export class FujiDrawUpdate {
  playerId: string;
  newCardId: string;
}


export class ActivePlayerUpdate {
  activePlayerId: string;
  discardedCards: FujiDiscardUpdate[];
}


export class GameVictoryUpdate {
  winningPlayers: string[];
}

export class FujiUpdate {
  playedCards: FujiPlayedCardUpdate[];
  discardedCards: FujiDiscardUpdate[];
  newDraws: FujiDrawUpdate[];
  activePlayerUpdate: ActivePlayerUpdate;
  gameVictoryUpdate: GameVictoryUpdate;
}
